src/components/TodosFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import classNames from 'classnames';
+import { FilterBy } from '../types/FilterOptions';
+
+type Props = {
+  filterBy: FilterBy,
+  setFilterBy: (option: FilterBy) => void;
+};
+
+export const TodosFilter: React.FC<Props> = ({ filterBy, setFilterBy }) => {
+  return (
+    <ul className="filters">
+      <li>
+        <a
+          href="#/"
+          onClick={e => {
+            e.preventDefault();
+            setFilterBy(FilterBy.All);
+          }}
```

Comment:
Pls move the function inside `onClick` handler to a separate const and use as a callback here
**Do the same for all such cases in this file!!!!**